olive/passes/onnx/hqq_quantization.py

```diff
@@ -142,6 +182,39 @@ def __init__(
         self.block_size = block_size
         self.axis = axis
 
+    def quantize_internal_numpy(self, b_ndarray):
+        """Convert numpy array to torch, quantize, and return numpy arrays."""
+        b_array_torch = torch.from_numpy(b_ndarray)
+        if torch.cuda.is_available():
+            b_array_torch = b_array_torch.cuda()
+
+        quant_weight_torch, scales_torch, zero_points_torch = self.quantize_internal(
+            b_array_torch.T, group_size=self.block_size
+        )
+        quant_weight_torch = quant_weight_torch.contiguous()
+        scales_torch = scales_torch.contiguous()
+        zero_points_torch = zero_points_torch.contiguous()
+
+        packed_torch = torch.zeros(
+            (quant_weight_torch.shape[0], quant_weight_torch.shape[1] // 2),
+            dtype=torch.uint8,
+            device=quant_weight_torch.device,
+        )
+        self.pack_on_row_fast_248bit(packed_torch, quant_weight_torch)
+        scales = scales_torch.cpu()
+        zero_points = zero_points_torch.cpu()
+
+        # reshape to the predefined shape in MatmulNbits
+        scales = scales.reshape(-1)
+        zero_points = zero_points.reshape(-1)
+        rows, cols = b_array_torch.shape
+        block_size = self.block_size
+        blob_size = block_size // 2
+        k_blocks = (rows + block_size - 1) // block_size
+        packed_torch = packed_torch.reshape(cols, k_blocks, blob_size)
+
+        return packed_torch.cpu(), scales, zero_points
```

Comment:
```suggestion
        scales = scales_torch
        zero_points = zero_points_torch

        # reshape to the predefined shape in MatmulNbits
        scales = scales.reshape(-1)
        zero_points = zero_points.reshape(-1)
        rows, cols = b_array_torch.shape
        block_size = self.block_size
        blob_size = block_size // 2
        k_blocks = (rows + block_size - 1) // block_size
        packed_torch = packed_torch.reshape(cols, k_blocks, blob_size)

        return packed_torch, scales, zero_points
```

ir.tensor can support gpu tensors